.circleci/dist_compile.yml

```diff
@@ -460,6 +460,7 @@ jobs:
     environment:
       VELOX_DEPENDENCY_SOURCE: SYSTEM
       simdjson_SOURCE: BUNDLED
+      DuckDB_SOURCE: BUNDLED
```

Comment:
Why are we using BUNDLED and not SYSTEM here. 

I believe this change (inadvertently) broken join fuzzer CI runs: https://github.com/facebookincubator/velox/issues/7943

CC: @kgpai